### DISCUSSION TITLE
How to avoid FOUC when using app directory

### DISCUSSION THREAD

**Discussion by MattHeslington** [nextauthjs/next-auth#6112] [Help]
Hello,

Great package!

I've implemented NextAuth in my project and it's working really well in terms of letting people signup etc.. but I'm experiencing FOUC with a protected page which I am protecting by using the following code:

```
import { redirect } from 'next/navigation'

export default async function Page() {
const user = await getCurrentUser()
    if (!user) {
        redirect(authOptions.pages.signIn)
    }
return (
   ...
)
```

The getCurrentUser function is as follows:

```
import { unstable_getServerSession } from 'next-auth/next'

import { authOptions } from 'lib/auth'

export async function getSession() {
    return await unstable_getServerSession(authOptions)
}

export async function getCurrentUser() {
    const session = await getSession()

    return session?.user
}
```

However, when an unauthorised user clicks on the link to visit this page, they get a FOUC before being redirected. A live example can be found [here](https://frontend-rebuild.vercel.app/) and by clicking on the 'Add you place' link. 

I have seen it working correctly with the NextJS app directory in the rather good [Taxonomy](https://github.com/shadcn/taxonomy) project, and have tried to implement the same strategy in my code.

Is there something obvious which I'm not doing, please?

Cheers, Matt

**Comment by iamcrisb** [comment]
Facing the same issue, did you manage to find a solution? Thx

**Reply by MattHeslington** [reply]
Not yet, will let you know if I do
